webapp/src/http_handlers/submit/mod.rs

```diff
@@ -0,0 +1,34 @@
+mod structs;
+
+use super::super::filework::*;
+use rocket::serde::json::Json;
+use compiler::data::input_data::{InputData, compiler_type::CompilerType};
+use compiler::handler::run_compilation;
+use std::path::PathBuf;
+
+#[post("/submit", format = "json", data = "<compilation_json>")]
+pub async fn post_submit(compilation_json: Json<structs::InputData>) 
+    -> Json<structs::OutputData>
+{
+    // TODO: respond with "internal server errors" explicitly?
+    // Input data validation
+    // TODO: validate input data: compilation options, language etc.
+
+    // Compilation goes here
+    let source_file = write_source_to_file(&compilation_json.code,
```

Comment:
Is it possible to create a separate function `try_to_compile(code, ops) -> Result<..>`?